tests/extras/datasets/avro/test_gbq_to_gcs_dataset.py

```diff
@@ -0,0 +1,119 @@
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+# or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from google.cloud.exceptions import NotFound
+
+from kedro.extras.datasets.avro import GBQTableGCSAVRODataSet
+from kedro.io import DataSetError
+
+DATASET = "dataset"
+TABLE_NAME = "table_name"
+PROJECT = "project"
+BUCKET = "bucket"
+OBJECT = "test.avro"
+
+
+@pytest.fixture
+def mock_bigquery_client(mocker):
+    return mocker.patch("google.cloud.bigquery.Client", autospec=True)
+
+
+@pytest.fixture
+def credentials():
+    return {"token": "foobar"}
+
+
+@pytest.fixture
+def gbq_to_gcs_avro_dataset(
+    credentials, load_args, save_args, mock_bigquery_client
+):  # pylint: disable=unused-argument
+    return GBQTableGCSAVRODataSet(
+        project=PROJECT,
+        dataset=DATASET,
+        table_name=TABLE_NAME,
+        bucket=BUCKET,
+        path_object=OBJECT,
+        credentials=credentials,
+        load_args=load_args,
+        save_args=save_args,
+    )
+
+
+class TestGBQTableGCSAVRODataSet:
+    def test_exists(self, mock_bigquery_client):
+        """Test `exists` method invocation."""
+        mock_bigquery_client.return_value.get_table.side_effect = [
+            NotFound("NotFound"),
+            "exists",
+        ]
+
+        data_set = GBQTableGCSAVRODataSet(PROJECT, DATASET, TABLE_NAME, BUCKET)
+        assert not data_set.exists()
+        assert data_set.exists()
+
+    @pytest.mark.parametrize(
+        "load_args", [{"use_avro_logical_types": False}], indirect=True
+    )
+    def test_load_extra_params(self, gbq_to_gcs_avro_dataset, load_args):
+        """Test overriding the default load arguments."""
+        for key, value in load_args.items():
+            assert gbq_to_gcs_avro_dataset._load_args[key] == value
+
+    @pytest.mark.parametrize(
+        "save_args", [{"use_avro_logical_types": False}], indirect=True
+    )
+    def test_save_extra_params(self, gbq_to_gcs_avro_dataset, save_args):
+        """Test overriding the default save arguments."""
+        for key, value in save_args.items():
+            assert gbq_to_gcs_avro_dataset._save_args[key] == value
+
+    def test_load(self, gbq_to_gcs_avro_dataset):
+        assert not gbq_to_gcs_avro_dataset.load()
+
+    def test_save(self, gbq_to_gcs_avro_dataset):
+        assert not gbq_to_gcs_avro_dataset.save()
+
+    def test_save_exceptions(self, gbq_to_gcs_avro_dataset):
```

Comment:
Can you divide this up into two tests each with a descriptive name of the exception that's expected to happen?